autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -1193,6 +1338,182 @@ async def get_pending_submissions(
     )
 
 
+async def get_admin_listings_with_versions(
+    status: prisma.enums.SubmissionStatus | None = None,
+    search_query: str | None = None,
+    page: int = 1,
+    page_size: int = 20,
+) -> backend.server.v2.store.model.StoreListingsWithVersionsResponse:
+    """
+    Get store listings for admins with all their versions.
+
+    Args:
+        status: Filter by submission status (PENDING, APPROVED, REJECTED)
+        search_query: Search by name, description, or user email
+        page: Page number for pagination
+        page_size: Number of items per page
+
+    Returns:
+        StoreListingsWithVersionsResponse with listings and their versions
+    """
+    logger.debug(
+        f"Getting admin store listings with status={status}, search={search_query}, page={page}"
+    )
+
+    try:
+        # Build the where clause for StoreListing
+        where_dict = {}
+        where_dict["isDeleted"] = False
+
+        sanitized_query = sanitize_query(search_query)
+        if sanitized_query:
+            # Find users with matching email
+            matching_users = await prisma.models.User.prisma().find_many(
+                where={"email": {"contains": sanitized_query, "mode": "insensitive"}},
+            )
+
+            user_ids = [user.id for user in matching_users]
+
+            # Set up OR conditions
+            where_dict["OR"] = [
+                {"slug": {"contains": sanitized_query, "mode": "insensitive"}},
+                {
+                    "Versions": {
+                        "some": {
+                            "name": {"contains": sanitized_query, "mode": "insensitive"}
+                        }
+                    }
+                },
+                {
+                    "Versions": {
+                        "some": {
+                            "description": {
+                                "contains": sanitized_query,
+                                "mode": "insensitive",
+                            }
+                        }
+                    }
+                },
+                {
+                    "Versions": {
+                        "some": {
+                            "subHeading": {
+                                "contains": sanitized_query,
+                                "mode": "insensitive",
+                            }
+                        }
+                    }
+                },
+            ]
+
+            # Add user_id condition if any users matched
+            if user_ids:
+                where_dict["OR"].append({"owningUserId": {"in": user_ids}})
+
+        # Calculate pagination
+        skip = (page - 1) * page_size
+
+        # Create proper Prisma types for the query
+        where = prisma.types.StoreListingWhereInput(**where_dict)
```

Comment:
Where are we not filtering the status of the `Versions` here, this is the most important.

Reply:
oh yeah that's a good place for it... twas a stressful last week lol